java/src/com/here/flexpolyline/PolylineEncoderDecoder.java

```diff
@@ -149,8 +159,7 @@ private void encodeHeader(int precision, int thirdDimensionValue, int thirdDimPr
                 throw new IllegalArgumentException("thirdDimPrecision out of range");
             }
 
-            if (thirdDimensionValue < 0 || thirdDimensionValue > 7 || thirdDimensionValue == 4 ||
-                thirdDimensionValue == 5) {
+            if (thirdDimensionValue < 0 || thirdDimensionValue > 7) {
```

Comment:
Why no longer check for RESERVED1 and RESERVED2?

It is arguably correct not to test for them here, but then there should be a second check that throws an `IllegalArgumentException("thirdDimensionValue has a reserved value")`

Reply:
Though it is reserved, it still valid to use. Some of the test case were using it without any effect. Previously there is no variable declared for these RESERVED case, to avoid NLP we had these checks. Now RESERVED case is handled so no need of those checks now.